app/controllers/settings/kubelet_compute_resources_reservations_controller.rb

```diff
@@ -0,0 +1,52 @@
+# Settings::KubeletComputeResourcesReservations is responsibe to manage all the requests
+# related to the kubelet compute resources reservations feature.
+class Settings::KubeletComputeResourcesReservationsController < SettingsController
+  def index
+    @kube_reservations = KubeletComputeResourcesReservation.find_or_initialize_by(
+      component: "kube"
+    )
+    @system_reservations = KubeletComputeResourcesReservation.find_or_initialize_by(
+      component: "system"
+    )
+    @eviction_hard = Pillar.find_or_initialize_by(pillar: "kubelet:eviction-hard")
+  end
+
+  def create
+    @kube_reservations = KubeletComputeResourcesReservation.find_or_initialize_by(
+      component: "kube"
+    )
+    @system_reservations = KubeletComputeResourcesReservation.find_or_initialize_by(
+      component: "system"
+    )
+
+    reservations = params["kubelet_compute_resources_reservations"]
+
+    @kube_reservations.cpu = reservations["kube_cpu"]
+    @kube_reservations.memory = reservations["kube_memory"]
+    @kube_reservations.ephemeral_storage = reservations["kube_ephemeral_storage"]
+
+    @system_reservations.cpu = reservations["system_cpu"]
+    @system_reservations.memory = reservations["system_memory"]
+    @system_reservations.ephemeral_storage = reservations["system_ephemeral_storage"]
+
+    @eviction_hard = Pillar.find_or_initialize_by(pillar: "kubelet:eviction-hard")
+
+    ActiveRecord::Base.transaction do
```

Comment:
could we also do most of the business logic in the model to keep the controller thin?